src/libraries/System.Formats.Tar/tests/TarReader/TarReader.File.Async.Tests.cs

```diff
@@ -308,6 +309,32 @@ public async Task SparseEntryNotSupportedAsync(string testFolderName, string tes
             await Assert.ThrowsAsync<NotSupportedException>(async () => await reader.GetNextEntryAsync());
         }
 
+        [Fact]
+        public async Task ReaderIgnoresFieldValueAfterTrailingNullAsync()
+        {
+            // Fields in the tar archives are terminated by a trailing null.
+            // When reading these fields the reader must ignore all bytes past that null.
+
+            // Construct an archive that has a filename with some data after the trailing null.
+            const string FileName = "  filename  ";
+            const string FileNameWithDataPastTrailingNull = $"{FileName}\0nonesense";
+            using MemoryStream ms = new();
+            using (TarWriter writer = new(ms, leaveOpen: true))
+            {
+                var entry = new UstarTarEntry(TarEntryType.RegularFile, FileNameWithDataPastTrailingNull);
+                writer.WriteEntry(entry);
+            }
+            ms.Position = 0;
+            // Check the writer serialized the complete name passed to the constructor.
+            bool archiveIsExpected = ms.ToArray().IndexOf(Encoding.UTF8.GetBytes(FileNameWithDataPastTrailingNull)) != -1;
```

Comment:
Thank you for ensuring that the null was written.  I wonder if we should truncate on write too - any opinion?  Of course that would make this test a bit harder to write (probably it would need to insert the null after writing).

Reply:
Ok, we can leave it as undocumented gray area then.